pkg/noderesourcetopology/filter.go

```diff
@@ -0,0 +1,263 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package noderesourcetopology
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+	v1qos "k8s.io/kubernetes/pkg/apis/core/v1/helper/qos"
+	bm "k8s.io/kubernetes/pkg/kubelet/cm/topologymanager/bitmask"
+	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
+	apiconfig "sigs.k8s.io/scheduler-plugins/pkg/apis/config"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	topoclientset "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/clientset/versioned"
+	topologyinformers "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/informers/externalversions"
+	listerv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/listers/topology/v1alpha1"
+)
+
+const (
+	// Name is the name of the plugin used in the plugin registry and configurations.
+	Name = "NodeResourceTopologyMatch"
+)
+
+var _ framework.FilterPlugin = &TopologyMatch{}
+
+type nodeTopologyMap map[string]topologyv1alpha1.NodeResourceTopology
+
+type PolicyHandler func(pod *v1.Pod, zoneMap topologyv1alpha1.ZoneList) *framework.Status
+
+type PolicyHandlerMap map[apiconfig.TopologyManagerPolicy]PolicyHandler
+
+// TopologyMatch plugin which run simplified version of TopologyManager's admit handler
+type TopologyMatch struct {
+	policyHandlers PolicyHandlerMap
+	lister         listerv1alpha1.NodeResourceTopologyLister
+	namespace      string
+}
+
+type NUMANode struct {
+	NUMAID    int
+	Resources v1.ResourceList
+}
+
+type NUMANodeList []NUMANode
+
+// Name returns name of the plugin. It is used in logs, etc.
+func (tm *TopologyMatch) Name() string {
+	return Name
+}
+
+// getTopologyPolicies return true if we're working with such policy
+func getTopologyPolicies(nodeTopology *topologyv1alpha1.NodeResourceTopology, nodeName string) []apiconfig.TopologyManagerPolicy {
+	policies := make([]apiconfig.TopologyManagerPolicy, 0)
+	for _, policy := range nodeTopology.TopologyPolicies {
+		policies = append(policies, apiconfig.TopologyManagerPolicy(policy))
+	}
+	return policies
+}
+
+func extractResources(zone topologyv1alpha1.Zone) v1.ResourceList {
+	res := make(v1.ResourceList)
+	for _, resInfo := range zone.Resources {
+		quantity, err := resource.ParseQuantity(resInfo.Allocatable.String())
```

Comment:
I guess you'd like to panic if err is not nil, so probably:

```go
quantity := resource.MustParse()
```

Reply:
no, I think it's better just to skip resource if it was specified incorrectly. As I know kube-apiserver returns BadRequest if yaml is not valid (resource name/value are also checked there)

Reply:
I think it should be handled the same way errors in the resource name are handled by the default scheduler i.e. api server does the validation. Lets double check in the scheduler code to see what happens in case the scheduler fails to parse the resources. Does it simply log and continues or panics and we follow the same in our plugin. Will look into the scheduler code to figure out the behavior.

Reply:
Discussed with @AlexeyPerevalov, since kube-apiserver doesn't apply pod with incorrect resource quantity we won't run into a situation where since kube-scheduler (default) gets it from kube-apiserver which means that in the scheduler plugin the resources in question have already been validated by API server and here too we won't run into a situation where we have invalid resourceNames. But wondering if there could be other reasons that the resource parsing could return an error?

Reply:
in case of NodeResourceTopology CRD both capacity and allocatable have a int-or-string type, one can put there incorrect quantity representation, like "10Gi10", I'm not sure it's possible to represent Quantity object in CRD to rely on verification in generated code, for apiserver it's just a string too. I don't think scheduler is a right place for verification.